src/mixins/name-request-mixin.ts

```diff
@@ -62,36 +64,39 @@ export default class NameRequestMixin extends Vue {
   }
 
   /**
-   * Checks if a name request response payload is consumable.
+   * Returns the Name Request's state.
    * @param nr the name request response payload
-   * @returns an object with result flags
    */
-  isNRConsumable (nr: any) : { isConsumable: boolean, expired: boolean, approved: boolean } {
-    // Ensure a name request payload is provided
+  getNrState (nr: any): NameRequestStates {
+    // Ensure a NR payload is provided.
     if (!nr) {
-      throw new Error('isNRConsumable() : no NR provided')
+      return null
     }
-    // Check if NR response has expired
-    if (new Date(nr.expirationDate) < new Date()) {
-      return { isConsumable: false, expired: true, approved: false }
+
+    // If the NR is expired, it is not consumable.
+    const expireDays = this.daysFromToday(nr.expirationDate)
+    if (isNaN(expireDays) || expireDays < 1) {
+      return NameRequestStates.EXPIRED
     }
 
-    // If the NR root state is not APPROVED, this NR is not consumable
+    // If the NR's root state is not APPROVED, it is not consumable.
     if (nr.state !== NameRequestStates.APPROVED) {
-      return { isConsumable: false, expired: false, approved: false }
+      return NameRequestStates.NOT_APPROVED
     }
 
-    // Check if the name request has already been consumed
-    let hasBeenConsumed = false
-    nr.names.forEach((name: { consumptionDate: any; }) => {
-      if (name.consumptionDate) {
-        hasBeenConsumed = true
-      }
-    })
-    if (hasBeenConsumed) {
-      return { isConsumable: false, expired: false, approved: true }
+    // NB: from here down, the NR is APPROVED (but maybe not consumable)
+
+    // If the NR has already been consumed, it is not consumable.
+    if (nr.names.some(name => name.consumptionDate)) {
+      return NameRequestStates.CONSUMED
+    }
+
+    // If the NR is awaiting consent, it is not consumable.
+    if (nr.consentFlag === false) {
```

Comment:
I think the state CONDITIONAL itself means that consent is required. From what I recall the `consentFlag == 'R'` when consent is received

Reply:
Thanks. I will look into it.

Reply:
You are right!

I updated the logic (in next commit) to check for Conditional state and consentFlag not received so that we can still display a "Need Consent" message.